packages/menu/src/MenuItem/MenuItem.styles.ts

```diff
@@ -160,6 +143,52 @@ export const getMenuItemContentStyles = ({
   `}
 `;
 
+interface NestedItemStyleArgs {
+  theme: Theme;
+  submenuDepth: number;
+  groupDepth: number;
+  submenuHasIcon: boolean;
+  groupHasIcon: boolean;
+}
+
+/** Styling for nested items */
+export const getNestedMenuItemStyles = ({
+  theme,
+  submenuDepth,
+  submenuHasIcon,
+  groupDepth,
+  groupHasIcon,
+}: NestedItemStyleArgs) => {
+  const submenuInset =
+    submenuDepth * (submenuHasIcon ? spacing[1000] : spacing[300]);
+  const groupInset = groupDepth * (groupHasIcon ? spacing[600] : spacing[300]);
+  const totalInset = submenuInset + groupInset;
+
+  return cx(
+    {
+      // The inset border for submenu items
+      [css`
+        &:after {
+          content: '';
+          position: absolute;
+          top: 0;
+          right: 0;
+          left: ${totalInset}px;
+          height: 1px;
+          background-color: ${menuColor[theme].border.default};
+        }
+      `]: submenuDepth > 0,
+    },
+    css`
+      .${inputOptionContentClassName} {
+        position: relative;
+        padding-left: ${totalInset}px;
+        border-top: 1px solid transparent;
```

Comment:
nit: should we use `rgba(255, 255, 255, 0)` instead of transparent?

Reply:
True... We don't transition this value, so it should be ok, but I'll use that for consistency